cmd/gocharm/gocharm.go

```diff
@@ -107,6 +107,9 @@ func buildCharm(p buildCharmParams) error {
 	if err := b.writeMeta(info.Relations); err != nil {
 		return errgo.Notef(err, "cannot write metadata.yaml")
 	}
+	if err := b.writeMetrics(); err != nil {
+		return errgo.Notef(err, "cannot write metrics.yaml")
+	}
```

Comment:
Might I suggest that instead of copying the metrics.yaml file from the package directory, we add a method to the Registry:

  func (r *Registry) RegisterMetric(name, kind, description string)

Then we can generate the metrics.yaml file automatically from that.